crates/typst-cli/src/main.rs

```diff
@@ -33,14 +35,21 @@ static ARGS: Lazy<CliArguments> = Lazy::new(CliArguments::parse);
 /// Entry point.
 fn main() -> ExitCode {
     let timer = Timer::new(&ARGS);
-    let res = match &ARGS.command {
+
+    let mut res = match &ARGS.command {
         Command::Compile(command) => crate::compile::compile(timer, command.clone()),
         Command::Watch(command) => crate::watch::watch(timer, command.clone()),
         Command::Query(command) => crate::query::query(command),
         Command::Fonts(command) => crate::fonts::fonts(command),
         Command::Update(command) => crate::update::update(command),
     };
 
+    // Leave the alternate screen if it was opened. This operation is done here
+    // so that it is executed prior to printing the final error.
+    res = res.or(terminal::out()
```

Comment:
This or confused me for a bit. Perhaps, this should just be `let res2 = terminal::out()...` and then `if let Err(msg) = res.or(res2)`.

Or maybe we can use expect here. If we can't leave the alternative screen, will print_error not also most probably just fail and then we get "failed to print error" rather than the more helpful "failed to leave alternate screen" panic?

Reply:
Also, should be `or_else` for laziness if we keep it.

Reply:
That was also my first thought, but I think it is intentionally not or_else because we want to leave the alternative screen even if `res` contains an error.

Reply:
I see, it's definitely confusing then.